pkg/storegateway/parquet_bucket_store.go

```diff
@@ -0,0 +1,266 @@
+package storegateway
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	"github.com/prometheus-community/parquet-common/convert"
+	"github.com/prometheus-community/parquet-common/schema"
+	"github.com/prometheus-community/parquet-common/search"
+	parquet_storage "github.com/prometheus-community/parquet-common/storage"
+	"github.com/prometheus-community/parquet-common/util"
+	"github.com/prometheus/prometheus/model/labels"
+	prom_storage "github.com/prometheus/prometheus/storage"
+	"github.com/thanos-io/objstore"
+	"github.com/thanos-io/thanos/pkg/block"
+	storecache "github.com/thanos-io/thanos/pkg/store/cache"
+	"github.com/thanos-io/thanos/pkg/store/hintspb"
+	"github.com/thanos-io/thanos/pkg/store/labelpb"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/cortexproject/cortex/pkg/util/validation"
+)
+
+type parquetBucketStore struct {
+	logger      log.Logger
+	bucket      objstore.Bucket
+	limits      *validation.Overrides
+	concurrency int
+
+	chunksDecoder *schema.PrometheusParquetChunksDecoder
+
+	matcherCache storecache.MatchersCache
+}
+
+func (p *parquetBucketStore) Close() error {
+	return p.bucket.Close()
+}
+
+func (p *parquetBucketStore) SyncBlocks(ctx context.Context) error {
+	return nil
+}
+
+func (p *parquetBucketStore) InitialSync(ctx context.Context) error {
+	return nil
+}
+
+func (p *parquetBucketStore) findParquetBlocks(ctx context.Context, blockMatchers []storepb.LabelMatcher) ([]*parquetBlock, error) {
+	if len(blockMatchers) != 1 || blockMatchers[0].Type != storepb.LabelMatcher_RE || blockMatchers[0].Name != block.BlockIDLabel {
+		return nil, status.Error(codes.InvalidArgument, "only one block matcher is supported")
+	}
+
+	blockIDs := strings.Split(blockMatchers[0].Value, "|")
+	blocks := make([]*parquetBlock, 0, len(blockIDs))
+	bucketOpener := parquet_storage.NewParquetBucketOpener(p.bucket)
+	noopQuota := search.NewQuota(search.NoopQuotaLimitFunc(ctx))
+	for _, blockID := range blockIDs {
+		block, err := p.newParquetBlock(ctx, blockID, bucketOpener, bucketOpener, p.chunksDecoder, noopQuota, noopQuota, noopQuota)
```

Comment:
Does it mean we don't apply any limits from parquet library today? But we still apply the existing store gateway limits, correct

Reply:
We can add new flags for later if needed.